storage/mempool.go

```diff
@@ -185,6 +185,45 @@ func (s *Storage) GetPendingUserState(stateID uint32) (*models.UserState, error)
 	}, nil
 }
 
+// does not return the full list of states, only the pending c2ts
+func (s *Storage) GetPendingUserStates(pubkey *models.PublicKey) ([]models.UserState, error) {
+	result := make([]models.UserState, 0)
+
+	maxUint32 := ^uint32(0)
+	pubKeyID := maxUint32
+
+	accounts, err := s.AccountTree.Leaves(pubkey)
+	if err == nil && len(accounts) > 0 {
+		// TODO: is this lookup really the right thing to do?
+		pubKeyID = accounts[0].PubKeyID
+	}
+
+	err = s.forEachMempoolTransaction(func(pendingTx *stored.PendingTx) error {
+		if pendingTx.TxType != txtype.Create2Transfer {
+			return nil
+		}
+
+		c2t := pendingTx.ToCreate2Transfer()
+		if c2t.ToPublicKey != *pubkey {
+			return nil
+		}
+
+		result = append(result, models.UserState{
+			PubKeyID: pubKeyID,
+			TokenID:  models.MakeUint256(0),
```

Comment:
Assuming WLD is TokenID 0 this will work, but shouldn't we fetch this from the fromStateId? (will get very complicated if that itself is pending)

Reply:
currently we only support a single token: https://github.com/worldcoin/hubble-commander/blob/e5f2708efb623f6292a585bcbd29eab6d1bfaed8/commander/executor/create_commitments.go#L249
-we only ever run batches using a WLD `feeReceiver` meaning all transactions with `tokenID`s other than 0 will fail

you're right that this eventually needs to be fixed though!